src/Generic/Data/Surgery/Internal.hs

```diff
@@ -866,6 +953,80 @@ type Arborify' arb op n nDiv2 f g =
    <=< SplitAt nDiv2
    ) (op f g)
 
+class GLazify f where
+  gLazify :: f x -> Lazify f x
+
+instance GLazify f => GLazify (M1 i m f) where
+  gLazify (M1 a) = M1 (gLazify a)
+
+instance (GLazify f, GLazify g) => GLazify (f :*: g) where
+  gLazify (a :*: b) = gLazify a :*: gLazify b
+
+instance (GLazify f, GLazify g) => GLazify (f :+: g) where
+  gLazify = \case
+    L1 a -> L1 (gLazify a)
+    R1 b -> R1 (gLazify b)
+
+instance GLazify (K1 i c) where
+  gLazify = id
+
+instance GLazify U1 where
+  gLazify = id
+
+instance GLazify V1 where
+  gLazify = id
+
+type family Lazify (f :: k -> *) :: k -> *
+type instance Lazify (M1 i m f) = M1 i (LazifyMeta m) (Lazify f)
+type instance Lazify (f :*: g) = Lazify f :*: Lazify g
+type instance Lazify (f :+: g) = Lazify f :+: Lazify g
+type instance Lazify (K1 i c) = K1 i c
+type instance Lazify U1 = U1
+type instance Lazify V1 = V1
```

Comment:
Regarding naming, I would say the functions themselves (`toLazyOR`, `Lazify`, etc.) are lazy, but the data is not. What `Lazify` does, rather than "make it lazy", is more of "remove decorations".

With that, one more consistent renaming would be `toLazyOR` → `toORLazy` (it's a lazy version of `toOR`), and `Lazify` → `UnDecorate`.

Of course, I'd also wonder how to keep the puns about surgery going, but if you don't want to go along with this silliness, it is also super fine to go for more serious names.

Here we're removing stuff so one of the `-ectomy` words might work, but all of them seem too long. Here it's really about changing the field selectors, so a made up word `SelEctomy` sounds good to me.

Reply:
I'm also keen on sticking to the theme as much as possible! `SelEctomy` is a nice idea (and `UnLazify`, if it is kept, could perhaps become `SelTransplant`?)

Regarding `toLazyOR` and `fromLazyOR` though, how should we go to make them easily discoverable?

Reply:
You can add a note to `fromOR` (is `toOR` actually also affected? I'm not sure.) that when inserting fields/constructors, there may be a mismatch with strict/unpacked fields, and you can switch to `fromLazyOR` in the case where the generic function doesn't care about it (which is all the existing ones I can think of, but you never know...).

Reply:
I just realized "lazy" actually made sense because the default annotations we insert "make the field lazy" (I was thinking of "lazy" in terms of "don't look at the annotations"). So I'm now thinking of switching back from `SelEctomy`... What do you think?

Reply:
Thanks, and sorry for the trouble!